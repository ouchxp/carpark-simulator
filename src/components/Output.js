import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';

const OutputContainer = styled.div`
  border: 1px solid lightgrey;
  border-radius: 4px;
  background-color: beige;
`;

const OutputEntry = styled.p`
  margin: 0 6px;
  margin-block-start: 2px;
  margin-block-end: 2px;
`;

const Output = ({ output }) => {
  // hide output area when nothing can be displayed
  if (!output.length) return null;
  return (
    <>
      <label>Output:</label>
      <OutputContainer>
        {output.map(x => (
          <OutputEntry key={x}>{x}</OutputEntry>
        ))}
      </OutputContainer>
    </>
  );
};

Output.propTypes = {
  output: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const mapStateToProps = state => ({
  output: state.app.output,
});

export default connect(mapStateToProps)(Output);
