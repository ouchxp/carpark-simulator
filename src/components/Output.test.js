import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Output from './Output';

const mockStore = configureStore([]);

describe('Output', () => {
  it('show all output', () => {
    const store = mockStore({
      app: { output: ['hello', 'world'] },
    });
    const wrapper = mount(
      <Provider store={store}>
        <Output />
      </Provider>
    );
    expect(wrapper.find('OutputEntry').get(0).props.children).toBe('hello');
    expect(wrapper.find('OutputEntry').get(1).props.children).toBe('world');
  });

  it('hide component when no output', () => {
    const store = mockStore({
      app: { output: [] },
    });
    const wrapper = mount(
      <Provider store={store}>
        <Output />
      </Provider>
    );
    expect(wrapper.find('OutputContainer').exists()).toBe(false);
  });

  it('renders Output that matches the snapshot', () => {
    const store = mockStore({
      app: { output: ['hello', 'world'] },
    });
    const wrapper = mount(
      <Provider store={store}>
        <Output />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
