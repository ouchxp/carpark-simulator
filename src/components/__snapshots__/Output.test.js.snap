// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output renders Output that matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Output)>
    <Output
      dispatch={[Function]}
      output={
        Array [
          "hello",
          "world",
        ]
      }
    >
      <label>
        Output:
      </label>
      <OutputContainer>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "OutputContainer-YbkxD",
                "isStatic": false,
                "lastClassName": "fyVKtg",
                "rules": Array [
                  "border:1px solid lightgrey;border-radius:4px;background-color:beige;",
                ],
              },
              "displayName": "OutputContainer",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "OutputContainer-YbkxD",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="OutputContainer-YbkxD fyVKtg"
          >
            <OutputEntry
              key="hello"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "OutputEntry-kaurUP",
                      "isStatic": false,
                      "lastClassName": "khLSnz",
                      "rules": Array [
                        "margin:0 6px;margin-block-start:2px;margin-block-end:2px;",
                      ],
                    },
                    "displayName": "OutputEntry",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "OutputEntry-kaurUP",
                    "target": "p",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <p
                  className="OutputEntry-kaurUP khLSnz"
                >
                  hello
                </p>
              </StyledComponent>
            </OutputEntry>
            <OutputEntry
              key="world"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "OutputEntry-kaurUP",
                      "isStatic": false,
                      "lastClassName": "khLSnz",
                      "rules": Array [
                        "margin:0 6px;margin-block-start:2px;margin-block-end:2px;",
                      ],
                    },
                    "displayName": "OutputEntry",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "OutputEntry-kaurUP",
                    "target": "p",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <p
                  className="OutputEntry-kaurUP khLSnz"
                >
                  world
                </p>
              </StyledComponent>
            </OutputEntry>
          </div>
        </StyledComponent>
      </OutputContainer>
    </Output>
  </Connect(Output)>
</Provider>
`;
